metrics/CII_Best_Practices.md

```diff
@@ -1,47 +1,97 @@
 
-## 5. Sample Implementation
+## 2. Use Cases
 
-## 6. Known Implementations
+Consumers of the badge can quickly assess which FLOSS projects are following best practices and as a result are more likely to produce higher-quality secure software. 
 
-## 7. Test Cases (Examples)
+## 3. Formula
 
-## 8. External References (Literature)
+GET Request to CII Best Practices API with a specific URL -> JSON response with status information for the repository
 
+## 4. Sample Filter and Visualization
 
+![](https://i.imgur.com/mSformz.png)
 
-----
-```markdown
-# {Name of Metric}
+## 5. Sample Implementation (JavaScript)
 
-## 1. Description
-A description of what the metric is and what it captures.
-The first few sentences have to match the description in the [metrics list](../activity-metrics-list.md).
+```
+const owner = "box"
+const repo = "bart"
+//used as a sample repository
+var request = new XMLHttpRequest();
+              function loader() {
+	            const basestr = owner + "/" + repo;
+	            const augURL = 'https://github.com/' + basestr;
+	            request.open('GET', 
+                       'https://bestpractices.coreinfrastructure.org/projects.json?pq=' + augURL, true);
+	            request.onload = function () {
+    	        var data = JSON.parse(this.response)[0];
+    	        if (data != undefined) {
+        	        //console.log('CII NAME: ' + data.name);
+        	        console.log(data)
+        	   }
+       };
+    }
+loader();
+request.send();
+```
 
-## 2. Use Cases
-Provide examples of how the metric might inform different stakeholders through use cases.
 
-## 3. Formula
-A generic formula (in pseudo code) to generate the metric.
+## 6. Known Implementations
 
-## 4. Sample Filter and Visualization
-Include a Sample Filter and Visualization (screenshot) of the metric from any implementation.
+See https://github.com/Nebrethar/risk-flask for an isolated sample.
 
-## 5. Sample Implementation
-An example implementation, for example a SQL or Elasticsearch query.
+## 7. Test Cases (Examples)
 
-## 6. Known Implementations
-Examples of where and how metric is used. (include links to dashboard or location where metric is visible or is talked about having been used).
+The input is an API call; GET request to :
+```
+https://bestpractices.coreinfrastructure.org/projects.json?pq= + ciiURL
+```
+Where ciiURL is the URL that CII has on file for the project. This value is often a GitHub URL
 
-## 7. Test Cases (Examples)
-Sample inputs (including contexts) and expected outputs for this metric. Implementers can test their implementations against these test cases. For quantitative metrics, this could include a static repository with known metric results, or just inputs and output. For qualitative metrics, this may be more difficult.
+The output is a set of values pertaining to CII pest practices.
+Here is a snippet of values from an API call regarding box/bart:

```

Comment:
Maybe we want to know what the "full load" looks like? Reference another document?

Reply:
@Nebrethar : Maybe just a link out per our call discussion / Risk Working group.